### PR TITLE
Add latex deletion

### LINKED ISSUE #448: Allow Author to Delete LaTeX
Probably the best way to do this is to re-format the LaTeX message so it contains a mention of the user who requested it. This will both help with moderation, and allow the bot to track who requested a LaTeX and easily delete it when they react with the 🗑️ emoji.

### PR DESCRIPTION
**Pull request by dnzc**
Made it possible for people to delete their sent latex messages if they react within 30 seconds with a red cross.

Resolves #448

### CODE REVIEW

**Comment by AnonGuy** [comment]
Closes #463.

**Comment by dnzc** [comment]
Okie, added that functionality.

File: cdbot/cogs/maths.py [hunk 1]
```diff
@@ -200,7 +201,24 @@ async def latex_render(self, ctx: Context, expression: str):
...
                 image_bytes.seek(0)
-                await ctx.send(file=File(image_bytes, filename="result.png"))
+
+                # send the resulting image and add a cross reaction
+                message = await ctx.send(file=File(image_bytes, filename="result.png"))
+                await message.add_reaction("❌")
+
+                # checks if the person who reacted was the original latex author and that they reacted with a cross
+                def should_delete(reaction: Reaction, user: Member):
+                    return ctx.message.author == user and reaction.emoji == "❌"
```
Comment (thebeanogamer) [inline cdbot/cogs/maths.py:211]:
```suggestion
                    return ctx.message.author == user and reaction.emoji == "🗑️"
```

File: cdbot/cogs/maths.py [hunk 2]
```diff
@@ -200,7 +201,24 @@ async def latex_render(self, ctx: Context, expression: str):
...
+
+                # if the latex author reacts with a cross within 30 secs of sending, delete the rendered image
+                # otherwise delete the cross reaction
+                try:
+                    await self.bot.wait_for("reaction_add", check=should_delete, timeout=30)
+                except asyncio.TimeoutError:
+                    await message.remove_reaction("❌", self.bot.user)
```
Comment (thebeanogamer) [inline cdbot/cogs/maths.py:218]:
```suggestion
                    await message.remove_reaction("🗑️", self.bot.user)
```

File: cdbot/cogs/maths.py (hunk 1 above)
Comment (dnzc) [reply cdbot/cogs/maths.py:211]: agreed

Comment (dnzc) [reply cdbot/cogs/maths.py]: Looks like committing only changed one of the emojis... I'll change them all then

**Pull request merged**